strings/ctype-mb.c

```diff
@@ -403,7 +403,8 @@ my_copy_fix_mb(CHARSET_INFO *cs,
   size_t fixed_length;
   size_t min_length= MY_MIN(src_length, dst_length);
 
-  well_formed_nchars= my_ci_well_formed_char_length(cs, src, src + min_length,
+  const char* end = (min_length == 0) ? src : src + min_length;
+  well_formed_nchars= my_ci_well_formed_char_length(cs, src, end,
```

Comment:
Sorry, but that is a crooked code... You should either return from function if min_length == 0 or assert min_length > 0. In all below similar cases you should not add app logic condition (len == 0) as this is not the app logic. In most cases these should be replaced by DBUG_ASSERT() (you may treat it as "in all cases", as this is refactoring-safe).

Reply:
In `CMAKE_BUILD_TYPE=RelWithDebInfo`, which I think what we should focus on, neither `assert()` nor `DBUG_ASSERT()` have any impact.